kroxylicious/src/test/java/io/kroxylicious/proxy/internal/FilterHandlerTest.java

```diff
@@ -52,6 +63,150 @@ public void testForwardRequest() {
         assertEquals(frame, propagated, "Expect it to be the frame that was sent");
     }
 
+    @Test
+    public void testDeferredRequestTimeout() throws Exception {
+        var filterFuture = new CompletableFuture<RequestFilterResult>();
+        ApiVersionsRequestFilter filter = (apiVersion, header, request, context) -> filterFuture;
+        buildChannel(filter, 50L);
+        writeRequest(new ApiVersionsRequestData());
+        Thread.sleep(60L);
```

Comment:
Can we wait for something to happen instead of just sleeping?

Reply:
Yes, it looks like embedded channel has `freezeTime` and `advanceTime` type methods on it.